superset-frontend/src/dashboard/util/charts/getFormDataWithExtraFilters.ts

```diff
@@ -25,16 +25,24 @@ import getEffectiveExtraFilters from './getEffectiveExtraFilters';
 const cachedFiltersByChart = {};
 const cachedFormdataByChart = {};
 
+interface GetFormDataWithExtraFiltersArguments {
+  chart: Record<string, any>;
```

Comment:
I'm wondering whether `chart` matches any existing types? If not, can we create a new interface in `src/dashboard/types.ts` and slowly expand it?

Reply:
I don't think we're at a point where it matches any existing types yet... My plan was to only start adding specific param types once needed to get typing to work. Then, as these were added, I'd expect other files to start failing without specific enough types. Finally, once everything is typed, we can go through and clean up unknowns, anys, and Records.

Thoughts?

Reply:
I think it'd be valuable to add specific typings as early as possible, so it can help future work down the road. The more specific the early typing is, the more it helps your second step.

If this doesn't work, maybe start with known props + `[key: string]: unknown`:

```ts
import { ChartProps } from '@superset-ui/chart';
import { chart } from 'src/chart/chartReducer';

type ChartReducerInitialState = typeof chart;

// chart query built from initialState
// Ref: https://github.com/apache/incubator-superset/blob/dcac860f3e5528ecbc39e58f045c7388adb5c3d0/superset-frontend/src/dashboard/reducers/getInitialState.js#L120
interface ChartQueryPayload extends ChartReducerInitialState {
  id: number;
  formData: ChartProps['formData'];
  form_data: ChartProps['RawFormData'];
  [key: string]: unknown;
}
```